src/osd/OSD.cc

```diff
@@ -4756,9 +4756,9 @@ void OSD::heartbeat()
 
 bool OSD::heartbeat_reset(Connection *con)
 {
+  std::lock_guard l(heartbeat_lock);
```

Comment:
According to https://en.cppreference.com/w/cpp/thread/lock_guard `std::lock_guard` was introduced in C++17?

I guess this won't work in mimic, and definitely not in luminous.

Reply:
use Mutex::Locker instead